nntrainer/layers/cl_layers/fc_layer_cl.cpp

```diff
@@ -0,0 +1,616 @@
+                                        unsigned int ldc,
+                                        RunLayerContext &context) {
+
+  bool result = false;
+
+  do {
+    result =
+      context.clCreateKernel(fc_sgemm_cl_kernel_, context.LayerKernel::FCSGEMM,
+                             FullyConnectedLayerCl::kernel_sgemm);
+    if (!result) {
+      break;
+    }
+
+    size_t m_size = sizeof(float) * M;
+    size_t n_size = sizeof(float) * N;
+    size_t k_size = sizeof(float) * K;
+    opencl::Buffer inputA(context.context_inst_, m_size * k_size, true,
+                          nullptr);
+
+    opencl::Buffer inputB(context.context_inst_, k_size * n_size, true,
+                          nullptr);
+
+    opencl::Buffer inOutC(context.context_inst_, m_size * n_size, true,
+                          nullptr);
+
+    result = inputA.WriteData(context.command_queue_inst_, A);
+    if (!result) {
+      break;
+    }
+
+    result = inputB.WriteData(context.command_queue_inst_, B);
+    if (!result) {
+      break;
+    }
+
+    result = inOutC.WriteData(context.command_queue_inst_, C);
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      0, &inputA, sizeof(cl_mem));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      1, &inputB, sizeof(cl_mem));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      2, &inOutC, sizeof(cl_mem));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      3, &M, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      4, &N, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      5, &K, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      6, &lda, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      7, &ldb, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    result = FullyConnectedLayerCl::kernel_sgemm.SetKernelArguments(
+      8, &ldc, sizeof(int));
+    if (!result) {
+      break;
+    }
+
+    const int work_groups_count[3] = {(int)M, 1, 1};
+    const int work_group_size[3] = {32, 32, 1}; // test-value
+
+    result = context.command_queue_inst_.DispatchCommand(
+      FullyConnectedLayerCl::kernel_sgemm, work_groups_count, work_group_size);
+    if (!result) {
+      break;
+    }
+
+    result = inOutC.ReadData(context.command_queue_inst_, C);
+    if (!result) {
+      break;
+    }
+
+  } while (false);
+}
+
+void FullyConnectedLayerCl::incremental_forwarding(RunLayerContext &context,
```

Comment:
It seems like incremental_forwarding does not use GPU. I'm worried little bit.. Currently we are using incremental_forwording for LLaMA Application. Could you check once again?

Reply:
Added GPU kernel call for `incremental_forwarding` as well in updated commit.